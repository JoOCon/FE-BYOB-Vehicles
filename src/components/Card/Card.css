.hidden {
  display: none;
}

.displayInfo {
  display: block;
}