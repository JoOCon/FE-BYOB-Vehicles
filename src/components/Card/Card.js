import React, { Component } from 'react';
import './Card.css';

class Card extends Component {
  constructor() {
    super();
    this.state = {
      displayInfo: false
    };
  }

  toggleInfo = () => {
    const { displayInfo } = this.state;

    this.setState({ displayInfo: !displayInfo })
  }

  render() {
    const { displayInfo } = this.state;
    const { ...model } = this.props;

    return (
      <article className="make-card-cont">
        <h2 onClick={this.toggleInfo}>
          {model.model_name}
          <div 
            className={displayInfo ? "display-model-info": "hidden"}
          >
            <p>
              Body: {model.body || 'null'}
            </p>
            <p>
              Engine: {model.engine || 'null'}
            </p>
            <p>
              Top Speed: {model.top_speed || 'null'}
            </p>
            <p>
              HP: {model.horse_power || 'null'}
            </p>
            <p>
              Transmission: {model.transmission || 'null'}
            </p>
          </div>
        </h2>
      </article>
    );
  }
}

export default Card;
