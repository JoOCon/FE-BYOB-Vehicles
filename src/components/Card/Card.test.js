import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card', () => {
  it('should match snapshot with makes', () => {
    const wrapper = shallow(<Card />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke deleteModel on button click', () => {
    const mockDeleteModel = jest.fn();
    const wrapper = shallow(<Card deleteModel={mockDeleteModel} />);
    const deleteX = wrapper.find('img');

    deleteX.simulate('click');

    expect(mockDeleteModel).toHaveBeenCalled();
  });
  describe('TOGGLEINFO', () => {
    it('should set state when invoked', () => {
      const wrapper = shallow(<Card />);
      expect(wrapper.state().displayInfo).toEqual(false);

      wrapper.instance().toggleInfo();

      expect(wrapper.state().displayInfo).toEqual(true);
    });
  });
});
