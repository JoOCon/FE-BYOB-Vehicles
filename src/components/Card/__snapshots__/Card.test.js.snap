// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot with makes 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          <span
            className="h2-span"
            onClick={[Function]}
          />
          <img
            alt="delete-btn"
            onClick={[Function]}
            src="delete.svg"
          />
        </h2>,
        <span
          className="hidden"
        >
          <p>
            Body: 
            null
          </p>
          <p>
            Engine: 
            null
          </p>
          <p>
            Top Speed: 
            null
          </p>
          <p>
            HP: 
            null
          </p>
          <p>
            Transmission: 
            null
          </p>
        </span>,
      ],
      "className": "model-card-cont",
      "id": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="h2-span"
              onClick={[Function]}
            />,
            <img
              alt="delete-btn"
              onClick={[Function]}
              src="delete.svg"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "h2-span",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "delete-btn",
              "onClick": [Function],
              "src": "delete.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Body: 
              null
            </p>,
            <p>
              Engine: 
              null
            </p>,
            <p>
              Top Speed: 
              null
            </p>,
            <p>
              HP: 
              null
            </p>,
            <p>
              Transmission: 
              null
            </p>,
          ],
          "className": "hidden",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Body: ",
                "null",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Body: ",
              "null",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Engine: ",
                "null",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Engine: ",
              "null",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Top Speed: ",
                "null",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Top Speed: ",
              "null",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "HP: ",
                "null",
              ],
            },
            "ref": null,
            "rendered": Array [
              "HP: ",
              "null",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Transmission: ",
                "null",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Transmission: ",
              "null",
            ],
            "type": "p",
          },
        ],
        "type": "span",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            <span
              className="h2-span"
              onClick={[Function]}
            />
            <img
              alt="delete-btn"
              onClick={[Function]}
              src="delete.svg"
            />
          </h2>,
          <span
            className="hidden"
          >
            <p>
              Body: 
              null
            </p>
            <p>
              Engine: 
              null
            </p>
            <p>
              Top Speed: 
              null
            </p>
            <p>
              HP: 
              null
            </p>
            <p>
              Transmission: 
              null
            </p>
          </span>,
        ],
        "className": "model-card-cont",
        "id": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="h2-span"
                onClick={[Function]}
              />,
              <img
                alt="delete-btn"
                onClick={[Function]}
                src="delete.svg"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "h2-span",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "delete-btn",
                "onClick": [Function],
                "src": "delete.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Body: 
                null
              </p>,
              <p>
                Engine: 
                null
              </p>,
              <p>
                Top Speed: 
                null
              </p>,
              <p>
                HP: 
                null
              </p>,
              <p>
                Transmission: 
                null
              </p>,
            ],
            "className": "hidden",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Body: ",
                  "null",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Body: ",
                "null",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Engine: ",
                  "null",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Engine: ",
                "null",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Top Speed: ",
                  "null",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Top Speed: ",
                "null",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "HP: ",
                  "null",
                ],
              },
              "ref": null,
              "rendered": Array [
                "HP: ",
                "null",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Transmission: ",
                  "null",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Transmission: ",
                "null",
              ],
              "type": "p",
            },
          ],
          "type": "span",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
